### PR TITLE
Finalize DG Implementation

### PR DESCRIPTION
**Pull request by lkldev**
This PR finalizes the DG Implementations that have been created. 
AB3 traces have been removed from DG as well.
Unused files in DG are removed.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -112,48 +112,94 @@ Here are the other classes in `Logic` (omitted from the class diagram above) tha
...
-
-<img src="images/BetterModelStudentClassDiagram.png" width="450" />
-
-<img src="images/BetterModelTaModuleClassDiagram.png" width="450" />
-
-<img src="images/BetterModelClassGroupClassDiagram.png" width="450" />
-
-<img src="images/BetterModelAssessmentClassDiagram.png" width="450" />
-
+* An `Entity` object is a generic object that can be used to represent instances of  `Student`/`TaModule`/`ClassGroup`/`Assessment`.
```
Comment (jxt00) [inline docs/DeveloperGuide.md:130]: Would like to clarify for the each of the associations below, would it be more appropriate to use a line without the diamond (i.e. `Student` is not destroyed when `TaModule` is destroyed, hence it is not a composition).
* `TaModule`: `Student`
* `ClassGroup`: `Student`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -348,9 +425,10 @@ Step 4. The user executes `mark attend c/1 w/1 s/1,2` to mark attendance for a l
 
 ![MarkUnmarkState1](images/MarkUnmarkState1.png)
```
Comment (jxt00) [inline docs/DeveloperGuide.md:426]: Are the `Attendance` objects missing?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (lkldev) [reply docs/DeveloperGuide.md:130]: Updated in the latest commit

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (lkldev) [reply docs/DeveloperGuide.md:426]: Updated in the latest commit

**Review by jxt00 — APPROVED** [review]
LGTM 💯

**Comment by lkldev** [comment]
Forgotten that I have an existing PR from this branch. @jxt00 can you take a look at the new commits for the UG again? Thanks

**Review by jxt00 — COMMENTED** [review]
UG line `225`: 
image

**Review by wxliong — APPROVED** [review]
LGTM! :)

**Pull request merged**